common/djangoapps/student/models.py

```diff
@@ -1589,3 +1589,19 @@ def user_can_skip_entrance_exam(cls, user, course_key):
             except EntranceExamConfiguration.DoesNotExist:
                 can_skip = False
         return can_skip
+
+
+class LanguageProficiency(models.Model):
+    """
+    Represents a user's language proficiency.
+    """
+    class Meta:
+        unique_together = (('code', 'user_profile'),)
+
+    user_profile = models.ForeignKey(UserProfile, db_index=True, related_name='language_proficiencies')
+    code = models.CharField(
```

Comment:
@jimabramson can answer this better than I. The `unique_together` call adds an index such that searching by language is still performant.